### ISSUE TITLE
[Search] a11y review

### ISSUE THREAD

**Issue by Westbrook** [adobe/spectrum-web-components#1420]
### Feedback

- Pick an accessible text for the reset button
- observed a strange behaviour:

### Path forward

1. Reviewing default with Spectrum, this also needs to be surfaced as an attribute of some sort of i18n.
2. What do you think should be happening here? While the input has focus, the screen reader cursor does not apply focus to its next stop on the page unless that stop "has a tab index", right? I experience that here: https://codepen.io/Westbrook/pen/BapeyLa

cc @jnurthen

**Comment by Westbrook** [comment]
The Spectrum team is suggesting "Clear search" for the reset button text.

**Comment by jnurthen** [comment]
2. I'm not sure of the question. Note that I only see the behaviour I observed in Safari - it works ok in Chrome. The difference is that in safari browser focus (and VO focus) seems to go back to the search field once the button is pressed, but in Chrome VO focus stays on the (now missing) "Clear search" button.

**Comment by Westbrook** [comment]
Let me revisit this, then. I felt I was seeing this as "expected" behavior related to the lack of focusability in the subsequent SR cursor location. If I'm still seeing it, I'll set some time to review together.

**Comment by Westbrook** [comment]
@jnurthen working through this, I want to confirm that there are two situations that you've brought up.

1. Above you discuss the location of the screen reader and focus cursor _after_ the "Clear search" button has been activated.
2. Previously you had outlined a scenario where the screen reader cursor progressed to a "disabled" input, but the focus cursor didn't.

### After Pressing Clear
For the first situation, while Chrome currently fails to relocation _both_ the screen reader and focus custom to the input after the clear button is activate, I believe that the Safari workflow of doing so should be considered correct here. This is what you see in the MDN docs for `input[type="search"]`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search While the "Clear search" button doesn't actually seem to accessible to the screen reader, after it is used via a pointing device the focus AND screen reader cursor are sent back to the now empty input.

If you still feel this is incorrect, can we work together to find the right way to document this as the way forward for Adobe? This will support our friends at React Spectrum in finding a path towards clarifying the same functionality in their library:
[![Example of a Search Field sending the focus cursor but not the screen reader cursor back to the input after clicking "Clear search"](https://user-images.githubusercontent.com/1156657/120004272-8dd4b000-bfa4-11eb-9db7-6afa263f0783.png)](https://react-spectrum.adobe.com/react-spectrum/SearchField.html#example)

### Moving the Screen Reader cursor to a disabled input (or otherwise non-focusable content)
This line of enquiry began with these steps as provided in your initial review:
- Focus on search field
- type something
- use ^⌥➡️ to go to clear button
- ^⌥space to press the clear button
- press ^⌥➡️ to go the next field (which is disabled)
- type - data is entered into the previous field

This appears to be the default functionality as provided by a similar pair of `input + input[disabled]` elements as outlined in https://codepen.io/Westbrook/pen/BapeyLa where the element after each focusable input is not focusable to while it receives the screen reader cursor it does not receive the focus cursor, which allows subsequent input to be made into an element that is not featured by the screen reader's cursor.

Please confirm whether you feel this continues to be an issue? It's possible that we could give focus to the non-focusable element in some way, but I'm remiss to fully investigate options there until I hear back from you.

**Comment by najikahalsema** [comment]
@jnurthen Could you review this and see if the a11y issues still apply?

**Comment by jnurthen** [comment]
I can't reproduce issue (2) any more

The label being changes to clear search would match react-spectrum.

**Comment by Westbrook** [comment]
@jnurthen thoughts on closing this in favor of the more recent review cycle?

**Comment by Westbrook** [comment]
Closing in preference of newer review.
